src/controllers/credentials/index.ts

```diff
@@ -24,28 +27,38 @@ export class CredentialsController extends HTMLController {
     const credentials: Credential[] = await this.cloudagent.getCredentials()
     req.log.info('retrieved credentials from a cloudagent %j', credentials)
 
-    const combined = credentials
-      .map((cred: Credential) => {
-        if (cred.credentialAttributes) {
-          return {
-            ...cred,
-            updated_at: new Date(),
-            company_name: cred?.credentialAttributes[0].name === 'company_name' && cred.credentialAttributes[0].value,
-            company_number:
-              cred.credentialAttributes[1].name === 'company_number' && cred.credentialAttributes[1].value,
-          }
-        }
+    const formatted: Credentials = this.format(credentials)
+    for (let i = 0; i < formatted.length; i++) {
+      const [connection]: ConnectionRow[] = await this.db.get('connection', {
+        agent_connection_id: formatted[i].connectionId,
       })
-      .filter((cred) =>
-        search !== '' ? cred?.company_name.toString().toLowerCase().includes(search.toLowerCase()) : true
-      )
-
-    req.log.info('%j', combined, credentials)
-    if (search !== '') {
-      req.log.info('applying search filter... %j', combined) // do some mapping
+      formatted[i].connection = connection
     }
 
+    formatted.filter((cred) => {
+      if (!cred.attributes) return false
+      const companyName = cred.attributes.map(({ name }: { name: string }) => {
+        return name === 'company_name'
+      })
+      if (search !== '') return true
+      req.log.info('checking if %s includes %s', companyName, search)
+
+      return companyName.toString().toLowerCase().includes(search.toLowerCase())
+    })
+
+    req.log.info('returming HTML along with formatted credentials %j', formatted)
+
     this.setHeader('HX-Replace-Url', search ? `/credentials?search=${encodeURIComponent(search)}` : `/credentials`)
-    return this.html(this.credentialsTemplates.listPage(combined as Credential[], search))
+    return this.html(this.credentialsTemplates.listPage(formatted, search))
+  }
+
+  private format(credentials: Credential[]): Credentials {
+    return credentials.map((cred) => {
+      return {
+        ...cred,
+        attributes: cred?.credentialAttributes || [].map(({ name, value }) => ({ [name]: value })),
```

Comment:
I think this should be

```
attributes: (cred?.credentialAttributes || []).map(({ name, value }) => ({ [name]: value })),
```

Note that at the moment you always map over an empty array?

Reply:
not always, even es-lint did not trip on this, also there is a unit test that checks if it renders conn ections

Reply:
ended up not wrapping in parenthesis, since it was working I did not change it